docs/Triggers.md

```diff
@@ -0,0 +1,100 @@
+Mock Firebolt: Triggers
+=======================
+
+# Overview
+
+You may define pre- and/or post- triggers that fire before and after Firebolt method calls are responded to by Mock Firebolt. These are "hooks" for you to add your own logic and/or actually alter or replace the response that would otherwise be returned by Mock Firebolt.
+
+When Mock Firebolt starts up, it will load any/all pre- and post- triggers defined in any of the directories passed via the `--trigger` command line argument(s). Then, when an app makes a Firebolt call and this call is sent to Mock Firebolt, Mock Firebolt looks to see if there is a pre- and/or post- trigger defined for that method call. If so, it invokes your JavaScript code right before and/or after it calculates the response it should return for that method call. 
+
+
+# Enabling Triggers
+
+By default, Mock Firebolt does not load/perform any triggers.
+
+To enable triggers, you must provide one or more directories which contain trigger definitions (JavaScript files) via one or more `--triggers` command line arguments when you start Mock Firebolt.
+
+Example:
+```
+cd server
+npm run dev -- --triggers ./src/triggers --triggers /some/other/dir/with/triggers
+```
+
+
+# File Organization
+
+Within each directory you pass via `--triggers` command line arguments, you must create subdirectories whose names match Firebolt method names (e.g., `lifecycle.ready`, `device.id`, etc.) and within these subdirectories you must create JavaScript files named `pre.mjs` and/or `post.mjs` depending on which trigger(s) you are defining.
+
+Your file system might look something like:
+
+```
+<trigger directory>
+  lifecycle.ready
+    pre.mjs
+    post.mjs
+  device.id
+    post.mjs
+  ...
+
+```
+
+
+# Trigger Definitions: JavaScript Function Definitions
+
+Within any `pre.mjs` file, you define a function named `pre`. Within any `post.mjs` file, you define a function named `post`. 
+
+These functions each take two parameters: a context object (see below) and a params array, which corresponds to the parameters passed by the app to the corresponding Firebolt method call.
+
+
+# The Context Object
+
+When Mock Firebolt invokes your triggers, it passes your function a context object as the first parameter that includes functions you can use (call) within your triggers.
+
+
+setTimeout
+: The standard setTimeout function; included for your convenience
+
+setInterval
+: The standard setInterval function; included for your convenience
```

Comment:
`setInterval` and `setTimeout` functions are already in the global namespace. Why do they need to be on `ctx` for convenience?